app/assets/javascripts/app_router.js

```diff
@@ -1,22 +1,48 @@
 var Pokedex = require("./models/pokedex.js");
 var PokedexView = require("./pokedex_view.js");
+var PokemonShowView = require("./pokemon_show_view.js");
 
-var AppRouter = function ($element) {
+var AppRouter = function (routes) {
   this.pokedex = new Pokedex();
-  this.pokedex.fetchThemAll(this.renderPokedex.bind(this));
-	this.$el = $element;
-  this.renderPokedex();
+  this.routes = routes;
+  this.container;
+};
+
+AppRouter.prototype.go = function (route) {
+  var regexRoute = /[\w-]+\//;
+  var routeName = route.match(regexRoute)[0];
+  var regexId = /\/(\d+)/;
+  var id = route.match(regexId) == null ? null : parseInt(route.match(regexId)[1]);
+  routeName = id ? routeName + ":id" : routeName;
+  var methodToCall = this.routes[routeName];
+  this[methodToCall](id);
 };
 
 AppRouter.prototype.renderPokedex = function () {
   var pokedexView = new PokedexView(this.pokedex);
+  this.pokedex.fetchThemAll(pokedexView.render.bind(pokedexView));
   this.renderView(pokedexView);
+  var self = this;
+  $("ul.pokedex-list").on("click", "a", function (event) {
```

Comment:
For further reference:

http://stackoverflow.com/questions/3666683/href-javascript-vs-href-javascriptvoid0